### PR TITLE
Added EndPoint.py

### PR DESCRIPTION
**Pull request by Jorissss**
Made a new pull request since some checks failed, hope its better now.

### CODE REVIEW

**Comment by Jorissss** [comment]
Since I'm new to contributing, I don't really understand the checks. What do they check? Especially the codecov checks I don't understand.

**Comment by cglewis** [comment]
@Jorissss the codecov checks are looking at test coverage of the patch you're submitting and the overall project as that patch is applied.  So in this case it's saying none of your PR has test coverage, and it brings the project's overall test coverage down by 1.65% if this was merged into the project.

codacy is complaining about a new issue that is introduced - if you click the details link next to it, should show you the problem you're introducing.

**Comment by Jorissss** [comment]
@cglewis Ah, I see, thanks!. So I should close this PR, add tests, fix the issue and try a new PR? Or is there some way to test this locally?

**Comment by cglewis** [comment]
the `Dockerfile` will run the tests, so if you have docker installed, from the repo directory where the `Dockefile` is you can run:
```
docker build -t poseidon .
docker run -it poseidon
```
and it should spit out the test results locally.

also you can just make commit updates to your branch and it will automatically update the existing PR.

**Comment by d-grossman** [comment]
to_json does not export state, next_state

File: poseidon/poseidonMonitor/EndPoint.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    def to_str(self):
+        '''make string representation of internals of object'''
+        strep = 'state: ' + self.state + '\n'
+        strep += 'next_state: ' + self.next_state + '\n'
+        strep += 'data: ' + str(self.data)
+        return strep
+
+    def to_json(self):
+        '''return a json view of the object'''
+        return json.dumps(self.data)
```
Comment (d-grossman) [inline poseidon/poseidonMonitor/EndPoint.py]: missing state and next_state
{'state':'s','next_state':'n','endpoint_data': {u'tenant': u'FLOORPLATE', u'mac': u'ac:87:a3:2b:7f:12', u'segment': u'prod', u'name': u'demo-laptop', u'ip-address': u'10.179.0.100'}}

File: poseidon/poseidonMonitor/endPoint.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+@author: Jorissss
+"""
+
+import hashlib
+import json
+
+
+
+class EndPoint:
+    def __init__(self, data):
```
Comment (d-grossman) [inline poseidon/poseidonMonitor/endPoint.py]: see write json

Comment (d-grossman) [reply poseidon/poseidonMonitor/endPoint.py]: looks like none of the asserts in test_endpoint_creation are passing circleci

File: poseidon/poseidonMonitor/EndPoint.py (hunk 1 above)
Comment (Jorissss) [reply poseidon/poseidonMonitor/EndPoint.py]: @d-grossman Ah, thought you wanted the json dump to look like the example in the issue, will change this.

File: poseidon/poseidonMonitor/endPoint.py (hunk 2 above)
Comment (Jorissss) [reply poseidon/poseidonMonitor/endPoint.py]: @d-grossman The error was that in the test_data the tags were not unicode, json.dumps does (apparently) encode in unicode so one of the to_str()'s came out with u's in front of the tags and one did not.

Comment (Jorissss) [reply poseidon/poseidonMonitor/endPoint.py]: @d-grossman codacy is also complaining about import pytest and the assert statements, anything I can do about that?

File: poseidon/poseidonMonitor/EndPoint.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+        for word in ['tenant', 'mac', 'segment', 'ip-address']:
+            pre_h = pre_h + str(self.data.get(str(word), 'missing'))
+        h.update(pre_h)
+        post_h = h.hexdigest()
+        return post_h
+
+    def to_str(self):
+        '''make string representation of internals of object'''
+        strep = 'state: ' + self.state + '\n'
```
Comment (d-grossman) [inline poseidon/poseidonMonitor/EndPoint.py]: Don't put in the newlines.

File: poseidon/poseidonMonitor/EndPoint.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+
+class EndPoint:
+    def __init__(self, data):
+        self.state = 'NONE'
+        self.next_state = 'NONE'
+        self.data = dict(data)
+
+    @staticmethod
+    def make_hask(self):
```
Comment (d-grossman) [inline poseidon/poseidonMonitor/EndPoint.py]: make_hash

File: poseidon/poseidonMonitor/endPoint.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        for word in ['tenant', 'mac', 'segment', 'ip-address']:
+            pre_h = pre_h + str(self.data.get(str(word), 'missing'))
+        h.update(pre_h.encode('utf-8'))
+        post_h = h.hexdigest()
+        return post_h
+
+    def to_str(self):
+        '''make string representation of internals of object'''
+        strep = 'state: ' + self.state + '\n'
+        strep += 'next_state: ' + self.next_state + '\n'
```
Comment (d-grossman) [inline poseidon/poseidonMonitor/endPoint.py]: Please remove the newlines.

File: poseidon/poseidonMonitor/test_endPoint.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+             u'segment': u'prod',
+             u'name': None,
+             u'ip-address': u'102.179.20.100'}
+
+def test_endpoint_creation():
+    endpoint1 = endPoint.EndPoint(test_data)
+    endpoint2 = endPoint.EndPoint.from_json(endpoint1.to_json())
+
+    assert endpoint1.make_hash() == endpoint2.make_hash()
+
```
Comment (d-grossman) [inline poseidon/poseidonMonitor/test_endPoint.py]: please add another test signature which the states are passed into the constructor.

something like:
endpoint1 = endPoint.EndPoint(test_data,state,next_state)

Comment (Jorissss) [reply poseidon/poseidonMonitor/test_endPoint.py]: @d-grossman when running 'docker build -f ./Dockerfile -t p2 .' it says all tests pass, when I run 'docker run p2' I get 'ValueError: invalid literal for int() with base 10: 'RABBIT_PORT'' on line 391 in poseidonMonitor.py.

Comment (d-grossman) [reply poseidon/poseidonMonitor/test_endPoint.py]: you should only need to do the first docker command (build) to see if tests pass.
I erroneously included the 2nd command, please disregard the output of the 2nd command.

**Comment by d-grossman** [comment]
circleci is still failing 

https://circleci.com/gh/CyberReboot/poseidon/568?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

it looks like 
endpoint1 = endPoint.EndPoint(test_data)
endpoint2 = endPoint.EndPoint.from_json(endpoint1.to_json())

endpoint1 is comprised of strings
endpoint2 is sometimes comprised of unicode strings

**Pull request merged**